go/indexer/services/l2/confirmed_headers.go

```diff
@@ -27,27 +29,57 @@ type ConfirmedHeaderSelector struct {
 	cfg HeaderSelectorConfig
 }
 
+func HeadersByRange(ctx context.Context, client *l2rpc.Client, startHeight uint64, count int) ([]*types.Header, error) {
```

Comment:
We can definitely dedupe these functions, the upstream RPC client should work well for both L2 geth and L1 execution clients

Reply:
Perhaps this is out of scope for this PR